Dockerfile

```diff
@@ -4,20 +4,26 @@ ENV S2I_GIT_VERSION="" \
     S2I_GIT_MAJOR="" \
     S2I_GIT_MINOR=""
 
+
+WORKDIR /tmp/source-to-image
+COPY . .
+
 ENV GOARCH="amd64"
 
-COPY . $GOPATH/src/github.com/openshift/source-to-image
+USER root
 
-RUN cd $GOPATH/src/github.com/openshift/source-to-image && \
-    make && \
-    install _output/local/bin/linux/${GOARCH}/s2i /usr/local/bin
+RUN make
 
 #
 # Runner Image
 #
 
 FROM registry.redhat.io/ubi8/ubi
 
-COPY --from=builder /usr/local/bin/s2i /usr/local/bin
+ENV GOARCH="amd64"
```

Comment:
Note that for the future we will want to remove this hard-code of GOARCH, as in the future S2I may need to run on other architectures (ex: Power, Z, ARM).

Reply:
Sure, thanks for the note.